api/types/container/hostconfig.go

```diff
@@ -270,33 +270,43 @@ type DeviceMapping struct {
 
 // RestartPolicy represents the restart policies of the container.
 type RestartPolicy struct {
-	Name              string
+	Name              RestartPolicyMode
 	MaximumRetryCount int
 }
 
+type RestartPolicyMode string
+
+const (
+	RestartPolicyDefault       RestartPolicyMode = ""
+	RestartPolicyDisabled      RestartPolicyMode = "no"
```

Comment:
We need to look if both `""` and `"no"` should actually be used; this looks to be a UX format that leaked into the API ("no" on the CLI side effectively should set "no policy"?)

Reply:
This does appear to be the case: the API docs say the following:

>  Enum: "" "no" "always" "unless-stopped" "on-failure"
>    Empty string means not to restart
>    no Do not automatically restart
>    always Always restart
>    unless-stopped Restart always except when the user has manually stopped the container
>    on-failure Restart only when the container exit code is non-zero


The CLI passes the string through as is; so I suspect that we can just coalesce these two cases together.